pandas/tests/tseries/offsets/test_dst.py

```diff
@@ -233,3 +235,30 @@ def test_nontick_offset_with_ambiguous_time_error(original_dt, target_dt, offset
     msg = f"Cannot infer dst time from {target_dt}, try using the 'ambiguous' argument"
     with pytest.raises(pytz.AmbiguousTimeError, match=msg):
         localized_dt + offset
+
+
+def test_series_dst_addition():
+    # GH#43784
+    startdates = pd.Series(
+        [
+            Timestamp("2020-10-25", tz="Europe/Berlin"),
+            Timestamp("2017-03-12", tz="US/Pacific"),
+        ]
+    )
+    offset1 = DateOffset(hours=3)
+    offset2 = DateOffset(days=1)
```

Comment:
looks like this only tests the DateOffset class, for which _use_relativedelta can be True.  but the code you changed affects a bunch of other classes

Reply:
Alright, I'll try and work on some tests for the other classes.